internal/libpcap/compile.go

```diff
@@ -37,18 +37,15 @@ const (
 	AvailableOffset
 )
 
-func CompileCbpf(expr string) (insts []bpf.Instruction, err error) {
+func CompileCbpf(expr string, l3 bool) (insts []bpf.Instruction, err error) {
 	if len(expr) == 0 {
 		return
 	}
 
-	/*
-		DLT_RAW linktype tells pcap_compile() to generate cbpf instructions for
-		skb without link layer. This is because kernel doesn't supply L2 data
-		for many of functions, where skb->mac_len == 0, while the default
-		pcap_compile mode only works for a complete frame data.
-	*/
-	pcap := C.pcap_open_dead(C.DLT_RAW, MAXIMUM_SNAPLEN)
+	pcap := C.pcap_open_dead(C.DLT_EN10MB, MAXIMUM_SNAPLEN)
```

Comment:
Nit: does the Go compiler optimize this statement? I.e., `pcap := C.pcap_open_dead(C.DLT_EN10MB, MAXIMUM_SNAPLEN)` is called only if `l3=false`?